src/refract/ExpandVisitor.cc

```diff
@@ -64,14 +64,23 @@ namespace refract
         };
 
         template <>
+        struct ExpandValueImpl<IElement*> {
+
+            template <typename Functor>
+            IElement* operator()(const IElement* value, Functor& expand) {
+                return expand(value);
+            }
+        };
+
+        template <>
         struct ExpandValueImpl<RefractElements> {
 
             template <typename Functor>
             RefractElements operator()(const RefractElements& value, Functor& expand) {
                 RefractElements members;
 
-                for (RefractElements::const_iterator it = value.begin() ; it != value.end() ; ++it) {
-                    members.push_back(expand(*it));
+                for (auto const& item: value) {
```

Comment:
Could be simplified using high-order transform?

```c++
const RefractElements members = std::transform(value.begin(), value.end(), expand);
```

Reply:
Seems to be good idea